pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -0,0 +1,328 @@
+// implements the documented semantics for Plants. updater is the UpdaterInterface used
+// to update the status of Plants.
+func NewDefaultPlantControl(k8sGardenClient kubernetes.Interface, recorder record.EventRecorder, config *config.ControllerManagerConfiguration, plantsLister gardencorelisters.PlantLister, secretLister kubecorev1listers.SecretLister) ControlInterface {
+	return &defaultPlantControl{
+		k8sGardenClient: k8sGardenClient,
+		plantLister:     plantsLister,
+		secretsLister:   secretLister,
+		recorder:        recorder,
+		config:          config,
+	}
+}
+
+func (c *defaultPlantControl) Reconcile(ctx context.Context, obj *gardencorev1alpha1.Plant, key string) error {
+	var (
+		plant  = obj.DeepCopy()
+		logger = logger.NewFieldLogger(logger.Logger, "plant", plant.Name)
+	)
+
+	if plant.DeletionTimestamp != nil {
+		return c.delete(plant, logger)
+	}
+
+	return c.reconcile(ctx, plant, key, logger)
+}
+
+func (c *defaultPlantControl) updatePlantConditions(plant *gardencorev1alpha1.Plant, conditions ...gardencorev1alpha1.Condition) (*gardencorev1alpha1.Plant, error) {
+	newPlant, err := kutil.TryUpdatePlantConditions(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(plant *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			plant.Status.Conditions = conditions
+			return plant, nil
+		})
+
+	return newPlant, err
+}
+
+func (c *defaultPlantControl) reconcile(ctx context.Context, plant *gardencorev1alpha1.Plant, key string, logger logrus.FieldLogger) error {
+	_, err := kutil.TryUpdatePlantStatusWithEqualFunc(c.k8sGardenClient.GardenCore(), retry.DefaultBackoff, plant.ObjectMeta,
+		func(p *gardencorev1alpha1.Plant) (*gardencorev1alpha1.Plant, error) {
+			if finalizers := sets.NewString(p.Finalizers...); !finalizers.Has(FinalizerName) {
+				finalizers.Insert(FinalizerName)
+				p.Finalizers = finalizers.UnsortedList()
+			}
+			return p, nil
+		}, func(cur, updated *gardencorev1alpha1.Plant) bool {
+			return sets.NewString(cur.Finalizers...).Has(FinalizerName)
+		})
+	if err != nil {
+		return err
+	}
+
+	var (
+		newConditions               = helper.MergeConditions(plant.Status.Conditions, helper.InitCondition(gardencorev1alpha1.PlantAPIServerAvailable), helper.InitCondition(gardencorev1alpha1.PlantEveryNodeReady))
+		conditionAPIServerAvailable = newConditions[0]
+		conditionEveryNodeReady     = newConditions[1]
+	)
+
+	kubeconfigSecret, err := c.secretsLister.Secrets(plant.Spec.SecretRef.Namespace).Get(plant.Spec.SecretRef.Name)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return c.updateStatusToUnknown(plant, "Referenced Plant secret could not be found.", conditionAPIServerAvailable, conditionEveryNodeReady)
+		}
+		return err
+	}
+	kubeconfig, ok := kubeconfigSecret.Data["kubeconfig"]
+	if !ok {
+		message := "Plant secret needs to contain a kubeconfig key."
+		return c.updateStatusToUnknown(plant, message, conditionAPIServerAvailable, conditionEveryNodeReady)
+	}
```

Comment:
We should also put a finalizer into the `secret` if we rely on it with `Plant`s.